src/uct/base/uct_md.c

```diff
@@ -250,43 +250,47 @@ static ucs_status_t uct_config_read(uct_config_bundle_t **bundle,
 }
 
 static uct_tl_component_t *uct_find_tl_on_md(uct_md_component_t *mdc,
+                                             uint64_t md_flags,
                                              const char *tl_name)
 {
     uct_md_registered_tl_t *tlr;
 
-    ucs_list_for_each(tlr, &mdc->tl_list, list) {
-        if (!strcmp(tl_name, tlr->tl->name)) {
-            return tlr->tl;
+    if ((tl_name == NULL) && (md_flags & UCT_MD_FLAG_SOCKADDR)) {
+        tlr = ucs_list_head(&mdc->tl_list, uct_md_registered_tl_t, list);
+        if (tlr == NULL) {
```

Comment:
this will not work, in case of empty list ucs_list_head() returns pointer to the list itself
need to write as follows
```
ucs_list_for_each(tlr, &mdc->tl_list, list) {
    if (((tl_name != NULL) && !strcmp(tl_name, tlr->tl->name)) ||
        ((tl_name == NULL) && (md_flags & UCT_MD_FLAG_SOCKADDR))) {
        return tlr->tl;
    }
}
return NULL;
```